src/main/java/com/scalar/db/storage/cassandra/ScannerImpl.java

```diff
@@ -43,7 +45,10 @@ public ScannerImpl(ResultSet resultSet, CassandraTableMetadata metadata) {
 
   @Override
   public Iterator<Result> iterator() {
-    return new ScannerIterator(resultSet, metadata);
+    if (scannerIterator == null) {
+      scannerIterator = new ScannerIterator(resultSet, metadata);
```

Comment:
I'm not fully sure but is it OK to cache an iterator?
If a user gets an iterator with `iterator()` and call `next()` several times and the call `iterator()` again,
doesn't the user expect a new iterator that hasn't moved the cursor?

Reply:
I think users expect that `iterator()` always returns an iterator whose cursor is reset to the first row as the general collections like ArrayList do. And currently, it looks like the behaviors between the adapters are different. Let's discuss it tomorrow.